ranger/ext/img_display.py

```diff
@@ -67,6 +87,47 @@ def move_cur(to_y, to_x):
     bin_stdout.write(tparm)
 
 
+def get_terminal_size():
+    farg = struct.pack("HHHH", 0, 0, 0, 0)
+    fd_stdout = sys.stdout.fileno()
+    fretint = fcntl.ioctl(fd_stdout, termios.TIOCGWINSZ, farg)
```

Comment:
This will give us the dimensions of the ncurses subwindow, right?

Reply:
> **TIOCGWINSZ**
>
> Fill in the winsize structure pointed to by the third argument with the screen width and height.
> 
> The winsize structure is defined in `sys/ioctl.h' as follows:
> 
>      	
> 
>     struct winsize
>     {
>       unsigned short ws_row;	/* rows, in characters */
>       unsigned short ws_col;	/* columns, in characters */
>       unsigned short ws_xpixel;	/* horizontal size, pixels */
>       unsigned short ws_ypixel;	/* vertical size, pixels */
>     };

Reply:
But is this the size of the terminal window or the ncurses subwindow/preview pane?

Reply:
This is the size of the terminal window. As a matter of fact, ncurses invokes the very same API under the hood on WinCh.

Reply:
Ah, right and then we use the max_cols and rows to limit the image size later, of course.